packages/sdk/src/modules/FundOperations.ts

```diff
@@ -29,32 +30,39 @@ export function withFundOperations<TBase extends MidasBaseConstructor>(Base: TBa
     }
 
     async approve(cTokenAddress: string, underlyingTokenAddress: string, amount: BigNumber) {
-      const token = getContract(underlyingTokenAddress, this.artifacts.EIP20Interface.abi, this.signer);
+      const token = getContract(
+        underlyingTokenAddress,
+        this.artifacts.EIP20Interface.abi,
+        this.signer
+      ) as EIP20Interface;
       const currentSignerAddress = await this.signer.getAddress();
 
       const hasApprovedEnough = (await token.callStatic.allowance(currentSignerAddress, cTokenAddress)).gte(amount);
```

Comment:
I always wanted to move this check from the sdk into the UI. It think it's ugly to check here. Would clean this up a lot.